platform/linuxbsd/detect.py

```diff
@@ -4,6 +4,11 @@
 from methods import get_compiler_version, using_gcc
 from platform_methods import detect_arch
 
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from SCons import Environment
+
```

Comment:
a Python file can be parsed in two different cases with two different parsers:
- when executed by Python (Python parser)
- when typechecked by mypy (mypy parser)

`TYPE_CHECKING` variable is set to `True` when mypy parser is in use and the condition `if TYPE_CHECKING:` is really the recommended way to distinguish the parser. 
The condition can be used for various reasons, usually to avoid needles imports on runtime, or the imported resources might not be installed/present on runtime. In this case I use the condition to make sure the script will be unaffected by the new import even in case the file is used in some other scenario than via `SCons`.

Why is the import/annotation needed in first place (to answer your comment "I'm not really convinced yet by the usability of type hints in Python") below. Type annotations are by default opt-in, so in order to check what's going on in the function, it needs to be annotated even if the annotation itself seems useless. When annotated, it can catch various kind of errors (like #66659) without any manual testing required.

Anyway, thanks for merging it.